include/utils.php

```diff
@@ -451,7 +451,7 @@ function getRunningUser()
     if ($runningUser == null) {  // matches null, false and ""
         if (is_windows()) {
             $runningUser = getenv('USERDOMAIN') . '\\' . getenv('USERNAME');
-        } else {
+        } elseif (function_exists('posix_getpwuid')) {
             $usr = posix_getpwuid(posix_geteuid());
             $runningUser = $usr['name'];
         }
```

Comment:
What happening in case when it's not a windows system and also posix_getpwuid is not exists too? I think it leads to an error somewhere.
Please handle the final `else` fork,  throw an exception and/or log the problem into suitecrm log so that the and user or caller can figure out what just happend here.

Reply:
@gymad I don't think any more checks are necessary, the function should just return `''` when all else fails. When I wrote this code I knew it wouldn't work on all systems, I just didn't predict (as I should have) that the missing Posix function would crash as a fatal on some systems, instead of just returning `null`. But it is ok for this function to return an empty string.

Reply:
@pgorod, @gymad doesn't this function already returns `null` if the posix function doesn't exist ?

Reply:
I think it does, but on some hostings it reacts violently, and instead of returning, FATALs out.

Reply:
Okey, let me explane what is the problem here from an different angle:
This function `getRunningUser()` called from many places, let see what happenning for e.g if it's a linux system and the `posix_getpwuid` function does not exists, so this function returns an empty string.
Here is the first case what I just found quickly:
cron.php:57: `$cronUser = getRunningUser();` it returns an empty string and the following code does not work properly, even if the `$sugar_config['cron']['allowed_cron_users']` filled with a correct user but also there is a fatal in the log: "cron.php: running as {$cronUser} is not allowed in allowed_cron_users in config.php. Exiting.". It's confusing, because the log note is not true also this code should not "Exiting".
Here is the second case:
install/performSetup.php:655: `addCronAllowedUser(getRunningUser());` It returns an empty string and passing to addCronAllowedUser() function, so this line does not do anything because `$addUser == ''` is true (see at include/utils.php:475). Also there is no any log note or error message meanwhile the `$sugar_config['cron']['allowed_cron_users']` filled with a correct user but the end-user have no clue what just happend, it seems a possible issue.
Third case: 
install/ready.php:281 `$webServerUser = getRunningUser();` returns an empty string so the help message bellove doesn't contains a correct user message on the screen. see at install/ready.php:286 `>sudo crontab -e -u '.$webServerUser.'</span><br>` - it's also confusing.
Fourth case: 
Scheduler/Scheduler.php:813 `$webServerUser = getRunningUser();` returns an empty string so the help message bellove doesn't contains a correct user message on the screen. Also confusing just like above.
And maybe there is some more other scenario when this change makes nonsense to the end-users causing a possible issue for someone in the future. I think it should be solved before I accept it.